### PR TITLE
[DROOLS-6746] make drools-core to depend on the static wiring

### PR DESCRIPTION
**Pull request by mariofusco**
JIRA: https://issues.redhat.com/browse/DROOLS-6746

Related PRs
https://github.com/kiegroup/drools/pull/4066
https://github.com/kiegroup/kogito-runtimes/pull/1782
https://github.com/kiegroup/kogito-examples/pull/1033

### CODE REVIEW

File: drools/kogito-drools/src/main/java/org/kie/kogito/drools/core/audit/KogitoWorkingMemoryFileLogger.java [hunk 1]
```diff
@@ -125,15 +123,15 @@ public void writeToDisk() {
             initializeLog();
         }
         try (FileOutputStream fileOut = new FileOutputStream(this.fileName + (this.nbOfFile == 0 ? ".log" : this.nbOfFile + ".log"), true);
-                Writer writer = new OutputStreamWriter(fileOut, StandardCharsets.UTF_8)) {
-            final XStream xstream = createTrustingXStream();
+                Writer writer = new OutputStreamWriter(fileOut, IoUtils.UTF8_CHARSET)) {
```
Comment (evacchi) [inline drools/kogito-drools/src/main/java/org/kie/kogito/drools/core/audit/KogitoWorkingMemoryFileLogger.java:126]: is this a copy/paste or...? (I mean the changes in this file are equivalent to the original)

**Review by evacchi — APPROVED** [review]

File: drools/kogito-drools/src/main/java/org/kie/kogito/drools/core/audit/KogitoWorkingMemoryFileLogger.java (hunk 1 above)
Comment (mariofusco) [reply drools/kogito-drools/src/main/java/org/kie/kogito/drools/core/audit/KogitoWorkingMemoryFileLogger.java:126]: Yes, this is a duplication of something that we also have in drools. I will check with a subsequent commit if I can get rid of this and other duplication and simply reuse what we have in drools.

File: jbpm/jbpm-flow/pom.xml [hunk 2]
```diff
@@ -68,7 +68,7 @@
       <exclusions>
         <exclusion>
           <groupId>org.drools</groupId>
-          <artifactId>drools-core</artifactId>
+          <artifactId>drools-wiring-dynamic</artifactId>
```
Comment (danielezonca) [inline jbpm/jbpm-flow/pom.xml:71]: Please create a follow-up ticket to review this usage, I expect we can get rid of drools-mvel

**Review by danielezonca — APPROVED** [review]
Just few minor comments

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/MissingDecisionTableDependencyError.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public MissingDecisionTableDependencyError() {
                 "Verify you have added decision table support to your project dependencies: \n" +
                 "\n" +
                 "<dependency>\n" +
-                "      <groupId>org.kie.kogito</groupId>\n" +
+                "      <groupId>org.drools</groupId>\n" +
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/MissingDecisionTableDependencyError.java]: the bom is in drools repository for main branch https://github.com/kiegroup/drools/blob/main/bom/drools-bom/pom.xml or am I missing something? @evacchi

**Pull request merged**